File: app/src/main/java/com/example/miniproject/ViewModel/ArticleViewModel.java
package com.example.miniproject.ViewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.miniproject.Model.ArticleModel;
import com.example.miniproject.Repository.ArticleRepository;

import java.util.List;

public class ArticleViewModel extends AndroidViewModel {

    private ArticleRepository articleRepository;

    public ArticleViewModel(@NonNull Application application) {
        super(application);
        this.articleRepository = new ArticleRepository(application);
    }

    public void callApiAndSaveInDB(String sourceId){
        articleRepository.getArticleListFromWebAndInsertInDB(sourceId);
    }

    public LiveData<List<ArticleModel>> getArticles() {
        return articleRepository.getArticleFromDb();
    }

    public LiveData<List<ArticleModel>> getArticles(String id) {
        return articleRepository.getArticleList(id);
    }

}

File: app/src/main/java/com/example/miniproject/viewModel/ArticleViewModel.kt
package com.example.miniproject.viewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.miniproject.model.ArticleModel
import com.example.miniproject.repository.ArticleRepository
import javax.inject.Inject

class ArticleViewModel @Inject constructor(private var articleRepository: ArticleRepository) : ViewModel() {

    fun callApiAndSaveInDB(sourceId: String?) {
        articleRepository.getArticleListFromWebAndInsertInDB(sourceId)
    }

    fun getArticles(id: String?): LiveData<List<ArticleModel?>?>? {
        return articleRepository.getArticleList(id)
    }

}